simulator/src/main/cpp/ChirpRadioPacket.cpp

```diff
@@ -0,0 +1,250 @@
+    this->sourceId = parse_2digit_int(source);
+
+    if(this->packetType == RADIO_ENQUIRE_PACKET)
+    {
+    }
+    else if(this->packetType == RADIO_RESPONSE_PACKET)
+    {
+        int i = 3;
+        int position = 0;
+        while(position < MAX_NEIGHBOUR_GROUPS_SIZE)
+        {
+            this->neighbourGroups[position] = parse_int(packet[i]);
+            i++;
+            position++;
+        }
+        this->neighbourGroupsSize = position;
+    }
+    else if(this->packetType == RADIO_DATA_PACKET)
+    {
+        this->groupId = parse_int(packet[3]);
+        char id[] = {packet[4], packet[5]};
+        this->id = parse_2digit_int(id);
+        char sequence_number[] = {packet[6], packet[7]};
+        this->sequenceNumber = parse_2digit_int(sequence_number);
+        this->lastPacket = parse_bool(packet[8]);
+        int i = 9;
+        int position = 0;
+        while(position < MAX_DATA_SIZE)
+        {
+            this->data[position] = packet[i];
+            i++;
+            position++;
+        }
+    } 
+    else
+    {
+        throw "ERROR: This is NOT a Valid Packet Type!";
+    }
+}
+
+int ChirpRadioPacket::getPacketType()
+{
+    return this->packetType;
+}
+
+void ChirpRadioPacket::setPacketType(int packetType)
+{
+    this->packetType = packetType;
+}
+
+int ChirpRadioPacket::getSourceId()
+{
+    return this->sourceId;
+}
+
+void ChirpRadioPacket::setSourceId(int sourceId)
+{
+    this->sourceId = sourceId;
+}
+
+int ChirpRadioPacket::getGroupId()
+{
+    if(this->packetType == RADIO_DATA_PACKET)
+    {
+        return this->groupId;
+    }
+}
+
+void ChirpRadioPacket::setGroupId(int groupId)
+{
+    if(this->packetType == RADIO_DATA_PACKET)
+    {
+        this->groupId = groupId;
+    }
+}
+
+int ChirpRadioPacket::getId()
+{
+    if(this->packetType == RADIO_DATA_PACKET)
+    {
+        return this->id;
+    }
+    return USHRT_MAX;
+}
+
+void ChirpRadioPacket::setId(int id)
+{
+    if(this->packetType == RADIO_DATA_PACKET)
+    {
+        this->id = id;
+    }
+}
+
+int ChirpRadioPacket::getSequenceNumber()
+{
+    if(this->packetType == RADIO_DATA_PACKET)
+    {
+        return this->sequenceNumber;
+    }
+}
+
+void ChirpRadioPacket::setSequenceNumber(int sequenceNumber)
+{
+    if(this->packetType == RADIO_DATA_PACKET)
+    {
+        this->sequenceNumber = sequenceNumber;
+    }
+}
+
+bool ChirpRadioPacket::isLastPacket()
+{
+    if(this->packetType == RADIO_DATA_PACKET)
+    {
+        return this->lastPacket;
+    }
+    return false;
+}
+
+void ChirpRadioPacket::setLastPacket(bool lastPacket)
+{
+    if(this->packetType == RADIO_DATA_PACKET)
+    {
+        this->lastPacket = lastPacket;
+    }
+}
+
+char* ChirpRadioPacket::getData()
+{
+    if(this->packetType == RADIO_DATA_PACKET)
+    {
+        return this->data;
+    }
+    return nullptr;
+}
+
+void ChirpRadioPacket::setData(char data[])
+{
+    if(this->packetType == RADIO_DATA_PACKET)
+    {
+        for(int i = 0; i < MAX_DATA_SIZE; i++)
```

Comment:
is there a reason you use a for loop here and a while loop for the bluetooth packet setData function?

Reply:
no reason... just did them on different days and i guess i was feeling while loopy that day... changed to match for code coherence